nomad/node_endpoint.go

```diff
@@ -1101,8 +1101,9 @@ func (n *Node) GetClientAllocs(args *structs.NodeSpecificRequest,
 
 // UpdateAlloc is used to update the client status of an allocation
 func (n *Node) UpdateAlloc(args *structs.AllocUpdateRequest, reply *structs.GenericResponse) error {
+	var err error
 	// Ensure the connection was initiated by another client if TLS is used.
-	err := validateTLSCertificateLevel(n.srv, n.ctx, tlsCertificateLevelClient)
+	err = validateTLSCertificateLevel(n.srv, n.ctx, tlsCertificateLevelClient)
```

Comment:
This is triggering a [semgrep rule](https://github.com/hashicorp/nomad/blob/main/.semgrep/rpc_endpoint.yml#L34), but the rule is being overly aggressive. We can merge without that, but we should probably follow-up by fixing the rule.

That being said, given that we rebind on `n.srv.forward` below, I'm not sure this gets us anything that having `var err error` in the loop below wouldn't?

Reply:
I'll clean this up tomorrow as well.